src/SIM/AdaptiveSetup.C

```diff
@@ -272,7 +285,13 @@ int AdaptiveSetup::calcRefinement (LR::RefineData& prm, int iStep,
   }
 
   prm.options.reserve(8);
-  prm.options.push_back(beta);
+  if (betaFunc) {
+    Vec4 X;
+    X.t = iStep-1;
+    prm.options.push_back((*betaFunc)(X));
+  } else
+    prm.options.push_back(beta);
+
   prm.options.push_back(knot_mult);
```

Comment:
If using `ScalarFunc` instead, then L288-295 could be replaced by the single line:
```
  prm.options.push_back(betaFunc ? (*betaFunc)(iStep-1) : beta);
```